tests/foreman/installer/test_installer.py

```diff
@@ -1663,3 +1663,81 @@ def test_positive_capsule_installer_logfile_check():
 
     :CaseAutomation: NotAutomated
     """
+
+
+@pytest.mark.destructive
+def test_installer_inventory_plugin_update(destructive_sat):
+    """DB consistency should not break after enabling the inventory plugin flags
+
+    :id: a2b66d38-e819-428f-9529-23bed398c916
+
+    :steps:
+        1. Enable the cloud inventory plugin flag
+
+    :expectedresults: inventory flag should be updated successfully without any db consistency
+        error.
+
+    :CaseImportance: Medium
+
+    :CaseLevel: System
+
+    :BZ: 1863597
+
+    :customerscenario: true
+
+    """
+    destructive_sat.create_custom_repos(rhel7=settings.repos.rhel7_repo)
+    command_output = destructive_sat.execute(
+        'satellite-installer --enable-foreman-plugin-rh-cloud '
+        '--foreman-proxy-plugin-remote-execution-ssh-install-key true'
+    )
+    assert 'Success!' in command_output.stdout
+    updated_cloud_flags = destructive_sat.execute(
+        'satellite-installer --help|grep "\'foreman_plugin_rh_cloud\' '
+        'puppet module (default: true)"'
+    )
+    assert 'true' in updated_cloud_flags.stdout
+    update_proxy_plugin_output = destructive_sat.execute(
+        'satellite-installer --help|grep -A1 '
+        '"foreman-proxy-plugin-remote-execution-ssh-install-key"'
+    )
+    assert 'true' in update_proxy_plugin_output.stdout
+
+
+@pytest.mark.destructive
+def test_installer_puppet_overload(destructive_sat):
+    """Check the puppet server tuning parameter update
+
+    :id: 70458b3a-1215-49ad-9c94-a115888c8dbd
+
+    :steps:
+        1. Set the new puppet server tuning parameter(puppet_server_jvm_param) in the puppet.
+        2. Set the value in array and it should be updated successfully.
+
+    :expectedresults: Puppet server tuning parameters should be reflected in the puppet-server
+        service
+
+    :CaseImportance: Medium
+
+    :CaseLevel: System
+
+    :BZ: 1920072
+
+    :customerscenario: true
+    """
+    puppet_server_jvm_param_1 = '-Djruby.logger.class=com.puppetlabs.jruby_utils.jruby.Slf4jLogger'
+    puppet_server_jvm_param_2 = '-XX:ReservedCodeCacheSize=1024m'
+    command_output = destructive_sat.execute(
+        f'satellite-installer --puppet-server-jvm-extra-args={puppet_server_jvm_param_1}'
```

Comment:
@devendra104 There is `InstallerCommand` helper in `robottelo/helpers.py` to construct satellite-installer commands, do you think we can use it in this test?

Reply:
Thanks, @jameerpathan111  It's a nice suggestion.